include/dali/core/geom/transform.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_GEOM_TRANSFORM_H_
+#define DALI_CORE_GEOM_TRANSFORM_H_
+
+#include "dali/core/geom/mat.h"
+
+namespace dali {
+
+DALI_HOST_DEV
+constexpr mat3 translation(vec2 offset) {
```

Comment:
Why no templates? I can imagine integer translation or operations with higher precision.

Reply:
Integer transforms make little sense. I was thinking about it in the context of higher precision. This variant won for being able to accept braced initializers without ambiguity.